### PR TITLE
Migrate question entity to RTK

### PR DESCRIPTION
**Pull request by ranquild**
Closes https://github.com/metabase/metabase/issues/40800

How to verify:
- CI is green

### CODE REVIEW

**Review by uladzimirdev — APPROVED** [review]

File: frontend/src/metabase-types/api/card.ts [hunk 1]
```diff
@@ -182,11 +182,46 @@ export type CardFilterOption =
...
+  id: CardId;
   ignore_view?: boolean;
 }
 
-export interface CardListQuery {
-  f?: CardFilterOption;
-  model_id?: CardId;
+export interface CreateCardRequest {
+  name: string;
+  dataset_query: DatasetQuery;
```
Comment (uladzimirdev) [inline frontend/src/metabase-types/api/card.ts:197]: do we duplicate Card type here?

Comment (ranquild) [reply frontend/src/metabase-types/api/card.ts:197]: It's not the same as `Card` - different fields

Comment (ranquild) [reply frontend/src/metabase-types/api/card.ts:197]: I could use something like `Pick<Card, "id"> & Pick<Partial<Card>, "name" | ...>`

Comment (ranquild) [reply frontend/src/metabase-types/api/card.ts:197]: Wdyt?

Comment (uladzimirdev) [reply frontend/src/metabase-types/api/card.ts:197]: no, not necessary atm, maybe we'll come up to some better type in the future. We have differences between FE and API fields and it's better to have fields duplicated I'd say

**Pull request merged**